collectd/conf.go

```diff
@@ -229,22 +238,22 @@ func validatedCollectdConfig(metrics *Metrics) (*collectdConf, error) {
 	}
 	for pipelineID, pipeline := range metrics.Service.Pipelines {
 		if strings.HasPrefix(pipelineID, "lib:") {
-			return nil, fmt.Errorf(`pipeline id prefix 'lib:' is reserved for pre-defined pipelines. Pipeline ID %q is not allowed.`, pipelineID)
+			return nil, reservedIdPrefixError("pipeline", pipelineID)
 		}
 		if len(pipeline.ReceiverIDs) != 1 {
 			return nil, errors.New("exactly one receiver id is required in the metrics service pipeline receiver id list.")
 		}
 		invalidReceiverIDs := findInvalid(definedReceiverIDs, pipeline.ReceiverIDs)
 		if len(invalidReceiverIDs) > 0 {
-			return nil, fmt.Errorf("metrics receivers not defined: %v", invalidReceiverIDs)
+			return nil, fmt.Errorf("metrics receiver %q from pipeline %q is not defined.", invalidReceiverIDs[0], pipelineID)
```

Comment:
In this error message, you switched from printing all of the invalid receiver ids, to just printing the first one.  Shouldn't we print out all of them?  What was the reason for the switch?

Reply:
This logic is nice in theory. In practice it always prints 1 message only, because only one receiver of `hostmetrics` is allowed. I changed the error message to align with the other failure cases.

How to provide a list of error messages instead of one is a more generic problem / improvement we need. That is better to be handled by another PR though. Ideally the validator should return as many errors in a list format to the users so that they can fix all the issues in one shot.